src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureFlagOptions.cs

```diff
@@ -10,14 +13,90 @@ namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.FeatureManage
     /// </summary>
     public class FeatureFlagOptions
     {
+        private SortedSet<string> _featureFlagPrefixes = new SortedSet<string>(Comparer<string>.Create((k1, k2) => -string.Compare(k1, k2, StringComparison.OrdinalIgnoreCase)));
```

Comment:
>- [](start = 115, length = 1)

What's the `-` for here? and is it the reason that [https://docs.microsoft.com/en-us/dotnet/api/system.stringcomparer.ordinalignorecase?view=net-5.0](https://docs.microsoft.com/en-us/dotnet/api/system.stringcomparer.ordinalignorecase?view=net-5.0) isn't being used?

Reply:
`-` is for sorting in descending order. I'm not sure what you mean by the second question.

Reply:
Why do we want to sort in descending order again?

`StringComparer.OrdinalIgnoreCase` is a built in string comparer so we don't have to have our own lambda.

Reply:
Consider two prefixes registered for trimming: "ABC" and "ABCD"
There might be some feature called "ABCDEF" and according to the registered prefixes, users would expects "EF" in their config. If the set is sorted in ascending order, we would just trim "ABC" and add "DEF" key to their config.

This means that we should try to match and trim the longest prefix first.

Reply:
I see. That's tricky, can you put a comment.

Reply:
Wow, 😄